ddtrace/contrib/jinja2/patch.py

```diff
@@ -0,0 +1,80 @@
+import jinja2
+from wrapt import wrap_function_wrapper as _w
+
+from ddtrace import tracer
+
+from ...ext import http
+from ...utils.wrappers import unwrap as _u
+
+DEFAULT_TEMPLATE_NAME = "<template>"
+
+
+def patch():
+    if getattr(jinja2, "__datadog_patch", False):
+        # already patched
+        return
+    setattr(jinja2, "__datadog_patch", True)
+    _w(jinja2, "environment.Template.render", _wrap_render)
+    _w(jinja2, "environment.Template.generate", _wrap_generate)
+    _w(jinja2, "environment.Environment.compile", _wrap_compile)
+    _w(jinja2, "environment.Environment._load_template", _wrap_load_template)
+
+
+def unpatch():
+    if not getattr(jinja2, "__datadog_patch", False):
+        return
+    setattr(jinja2, "__datadog_patch", False)
+    _u(jinja2.Template, "render")
+    _u(jinja2.Template, "generate")
+    _u(jinja2.Environment, "compile")
+    _u(jinja2.Environment, "_load_template")
+
+
+def _wrap_generate(func, instance, args, kwargs):
```

Comment:
We typically use `wrapped` instead of `func`. Seeing now that the `redis` integration is a little out date. My apologies there. Would you mind updating for these functions?

Reply:
We should add a check here (and before all wrapper functions):

```python
if not tracer.enabled:
    return wrapped(*args, **kwargs)
```

Just to make sure we are not generating traces if we shouldn't be.